torch/export/exported_program.py

```diff
@@ -1161,6 +1161,26 @@ def _update(
             verifiers=verifiers if verifiers is not None else self.verifiers,
         )
 
+    def _move_to(self, device: torch.device):
+        # change all the nodes with burnt-in device to the new device
+        for node in self.graph.nodes:
+            if "device" in node.kwargs:
+                kwargs = node.kwargs.copy()
+                kwargs["device"] = device
+                node.kwargs = kwargs
+        # move all the tensors and parameters to the new device
+        for k, v in self.state_dict.items():
+            if isinstance(v, torch.nn.Parameter):
+                self._state_dict[k] = torch.nn.Parameter(v.to(device))
+            else:
+                self._state_dict[k] = v.to(device)
+
+    def move_to(self, device: torch.device):
```

Comment:
can we make the API like
```
move_to(self, location: Union[torch.device, str, Dict[str, str]])
```
where `location` can be:
* torch.device
* string like "cpu" or "cuda"
* dictionary mapping existing device to the intended device
cc @zhxchen17 if you have more opinions

Reply:
@angelayi I'm not convinced that it makes sense to make this a method for now, can you make it a free function instead?